### PR TITLE
Fs create sighting

### PR DESCRIPTION
**Pull request by keqichen**

### CODE REVIEW

File: WhaleSpotting/Services/WhaleSightingService.cs [hunk 1]
```diff
@@ -1,25 +1,41 @@
...
 using WhaleSpotting.Models.Request;
 using WhaleSpotting.Repositories;
+using WhaleSpotting.Utilities;
 
 namespace WhaleSpotting.Services;
 
 public interface IWhaleSightingService
 {
    public WhaleSighting GetById(int id);
+
```
Comment (francescastocco) [inline WhaleSpotting/Services/WhaleSightingService.cs:11]: Extra line here

File: WhaleSpotting/Services/WhaleSightingService.cs [hunk 2]
```diff
@@ -1,25 +1,41 @@
...
     public WhaleSighting GetById(int id)
     {
         return _whaleSighting.GetById(id);
     }
+
+    public void CreateSighting(WhaleSightingRequest whaleSightingRequest, string authHeader)
+    {
+        var userName = AuthHelper.ExtractFromAuthHeader(authHeader).Username;
+        var ourUser = _users.GetByUsername(userName);
+        // User user = _users.GetById(6);
```
Comment (francescastocco) [inline WhaleSpotting/Services/WhaleSightingService.cs:37]: can we remove this comment?

File: WhaleSpotting/Services/WhaleSightingService.cs [hunk 3]
```diff
@@ -1,25 +1,41 @@
...
 
     public WhaleSighting GetById(int id)
     {
         return _whaleSighting.GetById(id);
     }
+
+    public void CreateSighting(WhaleSightingRequest whaleSightingRequest, string authHeader)
+    {
+        var userName = AuthHelper.ExtractFromAuthHeader(authHeader).Username;
+        var ourUser = _users.GetByUsername(userName);
```
Comment (francescastocco) [inline WhaleSpotting/Services/WhaleSightingService.cs:36]: Can we use the correct types here and rename the user variable?
```suggestion
        string userName = AuthHelper.ExtractFromAuthHeader(authHeader).Username;
        User user = _users.GetByUsername(userName);
```

**Review by francescastocco — CHANGES_REQUESTED** [review]
Looks good!

File: WhaleSpotting/Models/Request/WhaleSightingRequest.cs [hunk 4]
```diff
@@ -5,8 +5,8 @@ public class WhaleSightingRequest
     public DateTime DateOfSighting { get; set; }
     public float LocationLatitude { get; set; }
     public float LocationLongitude { get; set; }
-    public string PhotoImageURL { get; set; }
+    public string? PhotoImageURL { get; set; }
     public int NumberOfWhales { get; set; }
-    public string Description { get; set; }
-    public string WhaleSpecies { get; set; }
+    public string? Description { get; set; }
+    public string? WhaleSpecies { get; set; }
```
[the remark this answers is not in the document]
Comment (keqichen) [reply WhaleSpotting/Models/Request/WhaleSightingRequest.cs:11]: We want to make PhotoImageURL and Description optional, but not sure about  WhaleSpecies. We could discuss it with the group.

**Review by francescastocco — APPROVED** [review]
Nice!

**Pull request merged**